<head>
<title>MGP - The method of Merged Genotype Pairs for HIPP</title>
</head>
<body bgcolor="#FFFFCC">
<p align="center"><b><font color="#800000" size="+3">MGP</font></b></p>
<p align="center"><b><font color="#800000"><font size=+2>The method of Merged Genotype Pairs for HIPP problem </font></font></b></p> 
<p>MGP   is a   program to infer haplotype from genotype data using a heuristic algorithm which exploits the compatible relations  among genotypes to solve a reduced integer linear programming problem in  a smaller solution space.</p>
<p><strong><u><font color="#009900">How to use MGP?</font></u></strong> </p>
<p><font color="#FF0000"><strong><em>Note! Since the program MGP is linked with CPLEX 9.0 callable library, you have to setup up your PC environment before excute it. </em></strong></font></p>
<p>The steps for using MGP are as below.</p>
<ol>
  <li>Download the <a href="files/Windows/MGP.exe" target="_blank">executable file for Windows</a> (compiled by Visual C++, and linked with CPLEX 9.0  callable library).</li>
  <li>Usage: MGP <em>inputfile</em>, for example, MGP b2ar.txt </li>
</ol>
<p><u><strong><font color="#009900">The input file format</font></strong></u></p>
<p>Input file contains the   genotype data generated in experiment. <BR>
  The first line specifies the number of genotypes and SNPs, i.e. <em>genotype-number  SNP-number</em>.<BR>
One genotype is represented by a   consecutive sequence of SNP sites, and each SNP site can be 0 (for homozygous   wild type), 1 (for homozygous mutant), or 2 (for heterozygote). <br>
<A href="./files/input/b2ar.txt" target="_blank">A sample input file, b2ar.txt</A></p>
<p><u><strong><font color="#009900">The output file format</font></strong></u></p>
<p>There are three output files after excuting MGP; two of them are integer programming formulations for the corresponding input file, and one of them is the inferred configurations for the input genotypes.</p>
<p>The first integer programming formulation for <var>IP<sub>MPG1</sub>(C)</var> is called <em>MGP_inputfileStep1.lp</em>, for example, <a href="files/output/MGP_b2arStep1.lp" target="_blank">MGP_b2arStep1.lp</a> and the second integer programming formulation for <var>IP<sub>MPG2</sub>(C)</var> is called <em>MGP_inputfileStep2.lp</em>, for example, <a href="files/output/MGP_b2arStep2.lp" target="_blank">MGP_b2arStep2.lp</a>.</p>
<p>The inferred configurations for genotypes are output in file <em>MGP_inputfile.sol</em>.<BR>
  The first line contains: The number of selected haplotypes is: <em>haplotype-number</em><BR>
  One haplotype is represented by a consecutive sequence of SNP sites, and each SNP site can be either 0 (for homozygous wild type) or 1 (for homozygous mutant). Then resolutions of the input genotypes are listed.<br>
  The last line specifies the elapsed time for inferring the configurations for genotypes by GHI.<br>
  <A href="./files/output/MGP_b2ar.sol" target="_blank">A sample output file, MGP_b2ar.sol</A></p>
<p><u><strong><font color="#009900">Reference</font></strong></u></p>
<p>Wang, I. L. and Yang, H. E. (2006) Haplotyping populations by pure parsimony 
  based on compatible genotypes submitted to <em>Bioinformatics</em>.</p>
</body>